### ISSUE TITLE
compare error

### ISSUE THREAD

**Issue by MWangxj** [golang/go#33068]
Please answer these questions before submitting your issue. Thanks!

#### What did you do?
func TestReflect(t *testing.T) {
	if a!=-1 {
		fmt.Println("a")
		return
	}
	if a > 0 || a != -1 {
		fmt.Println(a>0)
		fmt.Println(a!=-1)
		fmt.Println(a)
	}
}

#### What did you expect to see?
nothing

#### What did you see instead?

false
false
-1
#### System details

```
go version go1.12.6 darwin/amd64
GOARCH="amd64"
GOBIN="/usr/local/gopath/bin"
GOCACHE="/Users/apple/Library/Caches/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/usr/local/gopath"
GOPROXY="https://goproxy.io"
GORACE=""
GOROOT="/usr/local/go1.12"
GOTMPDIR="/tmp"
GOTOOLDIR="/usr/local/go1.12/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/usr/local/gpath/test/go.mod"
GOROOT/bin/go version: go version go1.12.6 darwin/amd64
GOROOT/bin/go tool compile -V: compile version go1.12.6
uname -v: Darwin Kernel Version 18.5.0: Mon Mar 11 20:40:32 PDT 2019; root:xnu-4903.251.3~3/RELEASE_X86_64
ProductName:	Mac OS X
ProductVersion:	10.14.4
BuildVersion:	18E226
lldb --version: lldb-1001.0.13.3
  Swift-5.0
```

**Comment by clearcodecn** [comment]
```
package main

import (
	"fmt"
)


func main() {
	var (
		x int32 = -1
	)

	if x != -1 {
		fmt.Println("a")
		return
	}

	if x > 0 || x != -1 {
		fmt.Println(x > 0, x != -1, x > 0 || x != -1)
		fmt.Println("b")
		return
	}

	fmt.Println("c")
}

```

output:

```
false false false
b
```

**Comment by AlexRouSg** [comment]
Looks like #32560 which is fixed on `1.12.7`.
I've tested the code above and it does not repo.

**Comment by mvdan** [comment]
What @AlexRouSg said. In the future, please follow the entire issue template, including the "does this reproduce with the latest release" section.
